tests/test_github_caching.py

```diff
@@ -0,0 +1,47 @@
+from requests_mock import Mocker
+from howfairis.checker import Checker
+from howfairis.repo import Repo
+from howfairis.workarounds.github_caching import github_caching_check
+
+
+def initialize(requests_mock: Mocker, capsys):
+    owner = "fair-software"
+    repo_string = "howfairis"
+    filename = "README.rst"
+    url = f"https://github.com/{owner}/{repo_string}"
+    requests_mock.get(f"https://raw.githubusercontent.com/{owner}/{repo_string}/main/{filename}",
+                      json={}, status_code=200)
+    requests_mock.get(f"https://raw.githubusercontent.com/{owner}/{repo_string}/main/.howfairis.yml",
+                      json={}, status_code=200)
+    repo = Repo(url, branch="main")
+    checker = Checker(repo)
+    capsys.readouterr()
+    return(checker)
+
+
+def test_github_readme_update_check_critical_time(requests_mock: Mocker, capsys):
+    checker = initialize(requests_mock, capsys)
+    requests_mock.get(f"{checker.repo.api}/commits", json=[0], status_code=200)
```

Comment:
```suggestion
    requests_mock.get(f"{checker.repo.api}/commits", json=["nonzero array length should trigger warning"], status_code=200)
```

Reply:
This suggestion was committed via the command line